File: Apps/BankBusiness/backend/model/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.model;

[Table("payment")]
public class Payment
{
    [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id {get; set;}
    
    [Column("payment_account_id"), NotMapped]
    public PaymentAccount ProviderPaymentAccount {get; set;}
    
    [Column(name: "personal_account_id"), NotMapped]
    public object personal_account {get; set;}
    
    [Column(name: "transaction_id") , NotMapped]
    public object transaction {get; set;}
    
    [Column(name : "invoice_id")]
    public Invoice? Invoice {get; set;}
}

File: Apps/BankBusiness/backend/src/main/scala/model/Payment.scala
package model

import io.getquill.{InsertMeta, SchemaMeta, UpdateMeta, insertMeta, schemaMeta, updateMeta}

import java.util.UUID

case class Payment(
                  id: UUID,
                  paymentAccountId: Long,
                  invoiceId: UUID
                  )

object Payment {
  inline given SchemaMeta[Payment] = schemaMeta(
    "payment",
    _.id -> "id",
    _.paymentAccountId -> "payment_account_id",
    _.invoiceId -> "invoice_id"
  )

  inline given InsertMeta[Payment] = insertMeta(_.id)

  inline given UpdateMeta[Payment] = updateMeta(_.id)
}